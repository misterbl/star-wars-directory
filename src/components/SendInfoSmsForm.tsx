import React from "react";
import { FormikProps } from "formik";

interface ISendInfoSmsFormData {
  phoneNumber: string;
}
interface ISendInfoSmsForm {
  sendingInfoSms?: boolean;
  displaySentSuccess?: boolean;
  resetdisplaySentSuccess?: () => void;
}
class SendInfoSmsForm extends React.PureComponent<
  FormikProps<ISendInfoSmsFormData> & ISendInfoSmsForm
> {
  onPhoneNumberChange = (
    event: Pick<React.ChangeEvent<HTMLInputElement>, "target">
  ) => {
    this.props.setFieldValue("phoneNumber", event.target.value);
  };
  render() {
    const {
      values,
      handleSubmit,
      sendingInfoSms,
      displaySentSuccess,
      resetdisplaySentSuccess,
      errors,
      touched
    } = this.props;
    const phoneNumberError = Boolean(errors.phoneNumber && touched.phoneNumber);
    return (
      <form onSubmit={handleSubmit} className="ml-3 mt-3 mb-0 sms-form w-50">
        <label htmlFor="phoneNumber" className="text-white">
          Enter your number to get this information sent to you
        </label>
        <input
          placeholder="MOBILE NUMBER"
          autoComplete="off"
          id="phoneNumber"
          className="form-control"
          type="number"
          name="phoneNumber"
          value={values.phoneNumber}
          onFocus={resetdisplaySentSuccess}
          onChange={this.onPhoneNumberChange}
          aria-invalid={phoneNumberError}
        />
        {phoneNumberError && (
          <span
            id="errors-streetNumber"
            className="text-danger mt-1 d-block mb-4"
            role="alert"
          >
            {errors.phoneNumber}
          </span>
        )}
        {sendingInfoSms === false && displaySentSuccess && (
          <p className="text-success">message sent!</p>
        )}
      </form>
    );
  }
}
export default SendInfoSmsForm;
