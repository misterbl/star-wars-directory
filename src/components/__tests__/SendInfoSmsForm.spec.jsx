import * as React from "react";
import { shallow } from "enzyme";
import SendInfoSmsForm from "../SendInfoSmsForm";

describe("SendInfoSmsForm", () => {
  const props = {
    sendingInfoSms: false,
    values: {
      phoneNumber: "phoneNumber"
    },
    errors: {
      phoneNumber: false
    },
    touched: {
      phoneNumber: false
    },
    handleSubmit: jest.fn(),
    displaySentSuccess: false,
    resetDisplaySentSuccess: jest.fn(),
    setFieldValue: jest.fn()
  };
  const wrapper = shallow(<SendInfoSmsForm {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls setFieldValue on input change", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value: "value" } });
    expect(props.setFieldValue).toHaveBeenCalledWith("phoneNumber", "value");
  });
  it("calls handleSubmit on form submit", () => {
    const form = wrapper.find("form");
    form.simulate("submit");
    expect(props.handleSubmit).toHaveBeenCalled();
  });
  it("shows the error message if touched and error are true", () => {
    const propsWithError = {
      ...props,
      errors: {
        phoneNumber: true
      },
      touched: {
        phoneNumber: true
      }
    };
    const wrapperWithError = shallow(<SendInfoSmsForm {...propsWithError} />);
    const errorMessage = wrapperWithError.find("#errors-streetNumber");
    const input = wrapperWithError.find("input");
    input.simulate("change", { target: { value: "value" } });
    expect(errorMessage.length).toEqual(1);
  });
  it("shows the sms sent succes correctly", () => {
    const propsWithSucces = {
      ...props,
      displaySentSuccess: true
    };
    const wrapperWithSuccess = shallow(
      <SendInfoSmsForm {...propsWithSucces} />
    );
    const successMessage = wrapperWithSuccess.find(".text-success");
    expect(successMessage.length).toEqual(1);
  });
});
