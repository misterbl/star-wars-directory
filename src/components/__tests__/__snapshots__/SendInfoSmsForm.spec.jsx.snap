// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendInfoSmsForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SendInfoSmsForm
    displaySentSuccess={false}
    errors={
      Object {
        "phoneNumber": false,
      }
    }
    handleSubmit={[MockFunction]}
    resetDisplaySentSuccess={[MockFunction]}
    sendingInfoSms={false}
    setFieldValue={[MockFunction]}
    touched={
      Object {
        "phoneNumber": false,
      }
    }
    values={
      Object {
        "phoneNumber": "phoneNumber",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="text-white"
          htmlFor="phoneNumber"
        >
          Enter your number to get this information sent to you
        </label>,
        <input
          aria-invalid={false}
          autoComplete="off"
          className="form-control"
          id="phoneNumber"
          name="phoneNumber"
          onChange={[Function]}
          onFocus={[MockFunction]}
          placeholder="MOBILE NUMBER"
          type="number"
          value="phoneNumber"
        />,
        false,
        false,
      ],
      "className": "ml-5 mt-3 mb-0 sms-form w-50",
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter your number to get this information sent to you",
          "className": "text-white",
          "htmlFor": "phoneNumber",
        },
        "ref": null,
        "rendered": "Enter your number to get this information sent to you",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-invalid": false,
          "autoComplete": "off",
          "className": "form-control",
          "id": "phoneNumber",
          "name": "phoneNumber",
          "onChange": [Function],
          "onFocus": [MockFunction],
          "placeholder": "MOBILE NUMBER",
          "type": "number",
          "value": "phoneNumber",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
      false,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="text-white"
            htmlFor="phoneNumber"
          >
            Enter your number to get this information sent to you
          </label>,
          <input
            aria-invalid={false}
            autoComplete="off"
            className="form-control"
            id="phoneNumber"
            name="phoneNumber"
            onChange={[Function]}
            onFocus={[MockFunction]}
            placeholder="MOBILE NUMBER"
            type="number"
            value="phoneNumber"
          />,
          false,
          false,
        ],
        "className": "ml-5 mt-3 mb-0 sms-form w-50",
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter your number to get this information sent to you",
            "className": "text-white",
            "htmlFor": "phoneNumber",
          },
          "ref": null,
          "rendered": "Enter your number to get this information sent to you",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-invalid": false,
            "autoComplete": "off",
            "className": "form-control",
            "id": "phoneNumber",
            "name": "phoneNumber",
            "onChange": [Function],
            "onFocus": [MockFunction],
            "placeholder": "MOBILE NUMBER",
            "type": "number",
            "value": "phoneNumber",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
        false,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
